src/osd/ReplicatedPG.cc

```diff
@@ -6628,7 +6628,7 @@ void ReplicatedPG::finish_promote(int r, CopyResults *results,
     OpContext *tctx = repop->ctx;
     tctx->at_version = get_next_version();
     filter_snapc(tctx->new_snapset.snaps);
-    vector<snapid_t> new_clones(tctx->new_snapset.clones.size());
+    vector<snapid_t> new_clones;
```

Comment:
Adding a reserve() call will do what was intended by the improper constructor usage.

new.clones.reserve(tctx->new_snapset.clones.size());